File: apps/backend/content-service/src/main/java/com/misis/diplom/model/Course.java
package com.misis.diplom.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.UuidGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "courses")
public class Course {

  @Id
  @GeneratedValue
  @UuidGenerator
  private UUID id;

  private String title;

  private String description;

  private String imageUrl;

  @Enumerated(EnumType.STRING)
  private LevelEducation levelEducation;

  private String sphere;

  @CreationTimestamp
  private LocalDateTime createdAt;

  @UpdateTimestamp
  private LocalDateTime updatedAt;

  @Version
  private long version;

  // --- Поля для математической модели ---

  // Просмотры
  private long views;

  // Лайки
  private long likes;

  // Кол-во оценок
  private long ratingsCount;

  // Средняя оценка
  private double averageScore;

  // Результат модели
  private double popularity;
}

File: apps/web/types/course.ts
export enum LevelEducation {
  BEGINNER = "BEGINNER",
  INTERMEDIATE = "INTERMEDIATE",
  ADVANCED = "ADVANCED",
}

export const LevelEducationLabels: Record<LevelEducation, string> = {
  [LevelEducation.BEGINNER]: "Начальный",
  [LevelEducation.INTERMEDIATE]: "Средний",
  [LevelEducation.ADVANCED]: "Продвинутый",
};

export interface Course {
  id: string;
  title: string;
  description: string;
  imageUrl?: string | null;
  levelEducation: LevelEducation;
  sphere?: string;

  views: number;
  likes: number;
  ratingsCount: number;
  averageScore: number;
  popularity: number;

  isLiked: boolean;
}
